platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -159,26 +159,31 @@ private static void printDiagnosticCount(Log logger, String kind, int number) {
 
     @VisibleForTesting
     void buildProblem(Diagnostic<? extends JavaFileObject> diagnostic, ProblemSpec spec) {
-        spec.id(mapKindToId(diagnostic.getKind()), mapKindToLabel(diagnostic.getKind()), GradleCoreProblemGroup.compilation().java());
         spec.severity(mapKindToSeverity(diagnostic.getKind()));
+        addId(spec, diagnostic);
+        addContextualLabel(spec, diagnostic);
         addFormattedMessage(spec, diagnostic);
-        addDetails(spec, diagnostic);
         addLocations(spec, diagnostic);
     }
 
+    private static void addId(ProblemSpec spec, Diagnostic<? extends JavaFileObject> diagnostic) {
+        String kebabCode = diagnostic.getCode().replace('.', '-');
```

Comment:
This is a name property for the problem ID. A better name for the local variable would be `name` or `idName`.

Reply:
Agreed, I've changed it!